tests/remote_js/httpHandler.js

```diff
@@ -0,0 +1,78 @@
+const fetch = require('node-fetch');
+var config = require('config');
+
+class HTTP {
+	static verbose = config.verbose;
+
+	static async request(method, url, headers = {}, data = {}, auth = false) {
+		let options = {
+			method: method,
+			headers: headers,
+			follow: 0,
+			redirect: 'manual',
+			body: ["POST", "PUT", "PATCH", "DELETE"].includes(method) ? data : null
+		};
+
+		if (auth) {
+			options.headers.Authorization = 'Basic ' + Buffer.from(auth.username + ':' + auth.password).toString('base64');
+		}
+
+		if (config.verbose >= 1) {
+			console.log(`\n${method} ${url}\n`);
+		}
+
+		//Hardcoded for running tests against containers
+		const localIPRegex = new RegExp("172.16.0.[0-9][0-9]");
+		if (url.match(localIPRegex)) {
+			url = url.replace(localIPRegex, 'localhost');
+		}
+
+		await new Promise(resolve => setTimeout(resolve, 0));
```

Comment:
`node-fetch` has a [bug](https://github.com/node-fetch/node-fetch/issues/1735) that causes socket to hang while running mocha tests.
Temporary solution is to add this `await new Promise(resolve => setTimeout(resolve, 0));` in `httpHandler` before executing each api call